src/config.ts

```diff
@@ -265,8 +278,123 @@ export const defaultConfig: Config = {
   concat: {spacing: DEFAULT_SPACING}
 };
 
+const tab10 = scheme('tableau10');
+
+export const signalBasedConfig: Config = {
+  signals: [
+    {
+      name: 'defaultFont',
+      value: 'sans-serif'
+    },
+    {
+      name: 'fontSizes',
+      value: {
+        text: 11,
+        guideLabel: 10,
+        guideTitle: 11,
+        groupTitle: 13,
+        groupSubtitle: 12
+      }
+    },
+    {
+      name: 'colors',
+      value: {
+        blue: tab10[0],
+        orange: tab10[1],
+        red: tab10[2],
+        teal: tab10[3],
+        green: tab10[4],
+        yellow: tab10[5],
+        purple: tab10[6],
+        pink: tab10[7],
+        brown: tab10[8],
+        gray0: '#000',
+        gray1: '#111',
+        gray2: '#222',
+        gray3: '#333',
+        gray4: '#444',
+        gray5: '#555',
+        gray6: '#666',
+        gray7: '#777',
+        gray8: '#888',
+        gray9: '#999',
+        gray10: '#aaa',
+        gray11: '#bbb',
+        gray12: '#ccc',
+        gray13: '#ddd',
+        gray14: '#eee',
+        gray15: '#fff'
+      }
+    },
+    {
+      name: 'gridColor',
+      update: 'colors.gray13'
+    },
+    {
+      name: 'domainColor',
+      update: 'colors.gray8'
+    }
+  ],
+  mark: {color: {signal: 'colors.blue'}},
+  rule: {color: {signal: 'colors.gray0'}},
+  text: {
+    color: {signal: 'colors.gray0'},
+    font: {signal: 'defaultFont'}
+  },
+  style: {
+    'guide-label': {
+      fill: {signal: 'colors.gray0'},
+      font: {signal: 'defaultFont'},
+      fontSize: {signal: 'fontSizes.guideLabel'}
+    },
+    'guide-title': {
+      fill: {signal: 'colors.gray0'},
+      font: {signal: 'defaultFont'},
+      fontSize: {signal: 'fontSizes.guideTitle'}
+    },
+    'group-title': {
+      fill: {signal: 'colors.gray0'},
+      font: {signal: 'defaultFont'},
+      fontSize: {signal: 'fontSizes.groupTitle'}
+    },
+    'group-subtitle': {
+      fill: {signal: 'colors.gray0'},
+      font: {signal: 'defaultFont'},
+      fontSize: {signal: 'fontSizes.groupSubtitle'}
+    },
+    cell: {
+      stroke: {signal: 'gridColor'}
+    }
+  },
+  axis: {
+    domainColor: {signal: 'domainColor'},
+    gridColor: {signal: 'gridColor'},
+    tickColor: {signal: 'domainColor'}
+  },
+  legend: {
+    gradientStrokeColor: 'gridColor',
```

Comment:
we definitely re-use domainColor for both tick and domain by default, so it might make sense to define this `domainColor` signal, but I'm not sure about `gridColor`.